### PR TITLE
Rework input

### PR DESCRIPTION
**Pull request by X9VoiD**
Depends on #699 and #801

## Changes

- `Interpolator`  -> `IAsyncFilter`
- `IVirtualTablet` -> `IPressureHandler`
- `IAsyncFilter` is handled by `AsyncFilterHandler` inside `OutputMode`
  - Only one async filter will be active at any given time
  - Removed the need for filtering synthetic tablet report
- `AbsoluteOutputMode` and `RelativeOutputMode` common logic merged to `BaseOutputMode`
- Common configurations for `IOutputMode` moved to `IOutputMode.OutputModeConfig`
- AbsoluteMode and RelativeMode now uses the same UX for editing
  - `ResetTime` replaced by `AsyncFilterHandler.WithinDelayTolerance` (simplified logic from now-deleted `Limiter`)
- `IAbsolutePointer` and `IRelativePointer` replaced by `IPointer`
- and some other changes that I might have forgotten

## Benchmark
Pre-Rework
| Method |     Mean |    Error |   StdDev |
|------- |---------:|---------:|---------:|
| Output | 66.07 ns | 0.859 ns | 0.718 ns |

Rework
| Method |     Mean |    Error |   StdDev |
|------- |---------:|---------:|---------:|
| Output | 42.64 ns | 0.877 ns | 1.109 ns |

### CODE REVIEW

**Comment by X9VoiD** [comment]
CodeFactor bad :<

File: OpenTabletDriver.Plugin/Output/Async/AsyncFilterHandler.cs [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                }
+            }
+        }
+
+        public static bool WithinDelayTolerance(double reportMs, double reportMsAvg)
+        {
+            if (reportMsAvg < 3)
+                return reportMs < 3;
+            else
+                return reportMs < (reportMsAvg * 1.5);
```
Comment (AbstractQbit) [inline OpenTabletDriver.Plugin/Output/Async/AsyncFilterHandler.cs:107]: This function looks kinda sus
image
Maybe we can just simplify it to
```suggestion
            return (reportMs < 4.5f) | (reportMs < (reportMsAvg * 1.5));
```
?

File: OpenTabletDriver.Plugin/Output/RelativeOutputMode.cs [hunk 2]
```diff
@@ -1,126 +1,65 @@
...
-                        pressureHandler.SetPressure((float)tabletReport.Pressure / (float)Tablet.Digitizer.MaxPressure);
+            var pos = base.Transpose(report);
+            var deltaTime = stopwatch.Restart().TotalMilliseconds;
+            reportMsAvg += (deltaTime - reportMsAvg) * 0.1;
+            Vector2? delta;
 
-                    if (Transpose(tabletReport) is Vector2 position)
-                        Pointer.Translate(position);
-                }
+            if (lastPos.HasValue && pos.HasValue && !skipReport && AsyncFilterHandler.WithinDelayTolerance(deltaTime, reportMsAvg))
```
Comment (AbstractQbit) [inline OpenTabletDriver.Plugin/Output/RelativeOutputMode.cs:51]: In this case of just checking nullable's value lazy eval will only hurt performance by unnecessary branching
```suggestion
            if (lastPos.HasValue & pos.HasValue & !skipReport & AsyncFilterHandler.WithinDelayTolerance(deltaTime, reportMsAvg))
```

File: OpenTabletDriver.Plugin/HPETDeltaStopwatch.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ public HPETDeltaStopwatch(bool startRunning = true)
             start = isRunning ? internalWatch.Elapsed : default;
         }
 
-        public static TimeSpan RunitmeElapsed => internalWatch.Elapsed;
+        public TimeSpan RunitmeElapsed => internalWatch.Elapsed;
```
Comment (AbstractQbit) [inline OpenTabletDriver.Plugin/HPETDeltaStopwatch.cs:14]: `internalWatch` is static, there is no need for `HPETDeltaStopwatch` instance to query app uptime
```suggestion
        public static TimeSpan RunitmeElapsed => internalWatch.Elapsed;
```

**Review by AbstractQbit — COMMENTED** [review]
Some thoughts on what caughed my eye

File: OpenTabletDriver.Plugin/HPETDeltaStopwatch.cs (hunk 3 above)
Comment (X9VoiD) [reply OpenTabletDriver.Plugin/HPETDeltaStopwatch.cs:14]: Ah, this change was not intended. It was when I'm debugging a weird relative mode bug where it jumps even after skipping one, or two reports.

File: OpenTabletDriver.Plugin/Output/Async/AsyncFilterHandler.cs (hunk 1 above)
Comment (X9VoiD) [reply OpenTabletDriver.Plugin/Output/Async/AsyncFilterHandler.cs:107]: Is something like this fine?
```cs
            return reportMs < Math.Max(3, reportMsAvg * 1.5);
```

Comment (AbstractQbit) [reply OpenTabletDriver.Plugin/Output/Async/AsyncFilterHandler.cs:107]: Hmm, yeah, that looks even better

**Comment by X9VoiD** [comment]
Dropping the PR for now, but will keep the branch on my repo as reference.

**Pull request closed without merging**